### ISSUE TITLE
eth_estimate_gas: web3.exceptions.ContractLogicError: execution reverted: Failed to submit transaction: insufficient balance for transfer

### ISSUE THREAD

**Issue by lusharenck** [zksync-sdk/zksync2-python#16]
**There is no problem with the testnet, but there will be errors when using it on the mainnet**

```
rom eth_account import Account
from eth_account.signers.local import LocalAccount
from web3 import Web3
from zksync2.module.request_types import EIP712Meta

from zksync2.core.types import Token, EthBlockParams
from zksync2.module.module_builder import ZkSyncBuilder
from zksync2.provider.eth_provider import EthereumProvider


ZKSYNC_MAIN_URL = "https://mainnet.era.zksync.io"
ETH_TEST_URL = "hhttps://rpc.flashbots.net"
PRIVATE_KEY = "MYPRIVATEKEY"

def deposit(amount: float):

    zksync = ZkSyncBuilder.build(ZKSYNC_TEST_URL)
    eth_web3 = Web3(Web3.HTTPProvider(ETH_TEST_URL))
    account: LocalAccount = Account.from_key(PRIVATE_KEY)
    eth_provider = EthereumProvider(zksync, eth_web3, account)
    chain_id = zksync.zksync.chain_id
    wei_amount = Web3.to_wei(amount, "ether")
    eth_token = Token.create_eth()
    
   addres = Web3.to_checksum_address("my_address")
    l2_gas_limit = eth_provider._zksync_web3.zksync.eth_estimate_gas({
        'value': self.web3.to_hex(wei_amount),
        'from': address,
        'to': to_address,
        'data': '0x',
        'eip712Meta': EIP712Meta(gas_per_pub_data=0)
    })



if __name__ == "__main__":
    deposit(0.1)
```

network zksync era main

error msg:
```
Exception has occurred: ContractLogicError
execution reverted: Failed to submit transaction: insufficient balance for transfer
  File "D:\CodeProject\blockchain\blockchain_new\source_interaction\base\zksync_era.py", line 64, in zksync_era_deposit_for_eth
    l2_gas_limit = self.web3_other._zksync_web3.zksync.eth_estimate_gas({
web3.exceptions.ContractLogicError: execution reverted: Failed to submit transaction: insufficient balance for transfer

During handling of the above exception, another exception occurred:

  File "D:\CodeProject\blockchain\blockchain_new\source_interaction\base\zksync_era.py", line 74, in zksync_era_deposit_for_eth
    raise web3_exceptions.ContractLogicError(e)
  File "D:\CodeProject\blockchain\blockchain_new\source_interaction\base\web3_utils.py", line 125, in _inner
    return func(self, **params)
  File "D:\CodeProject\blockchain\blockchain_new\source_interaction\base\web3_utils.py", line 91, in _inner
    return func(self, *args, address=address, private_key=private_key, **kwargs)
  File "D:\CodeProject\blockchain\blockchain_new\source_interaction\base\web3_utils.py", line 66, in _inner
    result = func(self, *args, **kwargs)
  File "D:\CodeProject\blockchain\blockchain_new\source_interaction\base\zksync_era.py", line 110, in zksync_era_deposit_for_main
    action_result = self.zksync_era_deposit_for_eth(amount=amount, to_address=receiver,
  File "D:\CodeProject\blockchain\blockchain_new\source_interaction\base\zksync_era.py", line 121, in <module>
    ze.zksync_era_deposit_for_main(amount=0.02, from_=2, to_=2)
web3.exceptions.ContractLogicError: execution reverted: Failed to submit transaction: insufficient balance for transfer
```

![image](https://user-images.githubusercontent.com/62750932/231161559-28ac2d36-a804-4b05-841e-5a75277fa865.png)

Is this the estimated gas on the zksync era, or am I getting it wrong?

**I want to know which function is called by the following interface**
![image](https://user-images.githubusercontent.com/62750932/231162904-fcb09073-66d0-4010-ad8d-dff106196573.png)

**Comment by danijelTxFusion** [comment]
Gas estimation on mainnet does not work. Temporary solution is to estimate gas on testnet and set `gasLimit` value for transactions that needs to be broadcasted to mainnet.

**Comment by danijelTxFusion** [comment]
@mixbe 

You are write when it comes to `gas_limit` it should be named `l2_gas_limit` (your PR that makes those changes is already merged). Also when `gas_limit` is not provided, the fixed default value of `10_000_000` is used. This not correct anymore and the gas limit should be estimated using [zks_estimateGasL1toL2](https://era.zksync.io/docs/api/api.html#zks-estimategasl1tol2). The provider does not have this RPC method implemented yet, but it will be implemented in the upcoming release.

**Comment by BoboTiG** [comment]
Any news, or plan to share with us? :)

**Comment by danijelTxFusion** [comment]
The issues are resolved in version `1.0.0`. To use the SDK, please refer to the [documentation](https://docs.zksync.io/build/sdks/python/getting-started.html) and [examples](https://github.com/zksync-sdk/zksync2-examples/tree/main/python).
